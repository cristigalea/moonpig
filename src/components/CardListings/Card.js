import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import LoadedImage from '../shared/LoadedImage';
import './styles.css';

const Card = ({ url, id }) => (
    <Link to={`/cards/${id}`} className="card">
        <LoadedImage url={url} />
    </Link>
);

Card.propTypes = {
    url: PropTypes.string.isRequired,
    id: PropTypes.string.isRequired
};

export default Card;