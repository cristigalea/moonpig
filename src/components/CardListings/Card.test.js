import React from 'react';
import {shallow} from 'enzyme';

import Card from './Card';

describe('<Card />', () => {
    it('should match the snapshot', () => {
        const component = shallow(<Card url="https://cdn.britannica.com/s:700x450/01/115901-004-AC4BD3CC.jpg" id="1" />);
        expect(component).toMatchSnapshot();
    });
});