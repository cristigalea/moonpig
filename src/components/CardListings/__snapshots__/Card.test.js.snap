// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    id="1"
    url="https://cdn.britannica.com/s:700x450/01/115901-004-AC4BD3CC.jpg"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <LoadedImage
        url="https://cdn.britannica.com/s:700x450/01/115901-004-AC4BD3CC.jpg"
      />,
      "className": "card",
      "replace": false,
      "to": "/cards/1",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "url": "https://cdn.britannica.com/s:700x450/01/115901-004-AC4BD3CC.jpg",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <LoadedImage
          url="https://cdn.britannica.com/s:700x450/01/115901-004-AC4BD3CC.jpg"
        />,
        "className": "card",
        "replace": false,
        "to": "/cards/1",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "url": "https://cdn.britannica.com/s:700x450/01/115901-004-AC4BD3CC.jpg",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
